gcc/rust/checks/errors/rust-unsafe-checker.cc

```diff
@@ -262,10 +262,23 @@ UnsafeChecker::visit (MethodCallExpr &expr)
 void
 UnsafeChecker::visit (FieldAccessExpr &expr)
 {
-  // FIXME: If the receiver is an union, we need to be in an unsafe context to
-  // access it. Make sure to check.
-
   expr.get_receiver_expr ()->accept_vis (*this);
+
+  if (is_unsafe_context ())
+    return;
+
+  TyTy::BaseType *receiver_ty;
+  context.lookup_type (expr.get_receiver_expr ()->get_mappings ().get_hirid (),
+		       &receiver_ty);
+
+  if (receiver_ty->get_kind () == TyTy::TypeKind::ADT)
+    {
+      auto maybe_union = static_cast<TyTy::ADTType *> (receiver_ty);
+      if (maybe_union->get_adt_kind () == TyTy::ADTType::ADTKind::UNION)
```

Comment:
we have a helper on the ADTType is_union which would be best to use here.

```
const auto& adt = static_cast<const TyTy::ADTType&>(*receiver_ty);
if (adt.is_union()) {
   error_message;
}
```

Reply:
Fixed in [f532ae5](https://github.com/Rust-GCC/gccrs/pull/1416/commits/f532ae5aa37c8555dd6a99d8502436509db7930f)